File: 1446-consecutive-characters/1446-consecutive-characters.c
#define MAX(x,y) (x > y ? x : y)

int maxPower(char * s){
    int i = 0, j = 0, sum = 0;
    while (s[i])
    {
        if (s[i] == s[i + 1])
        {
            while (s[i] == s[i + 1])
                i++;
            sum = MAX(sum, (i - j));
        }
        i++;
        j = i;
    }
    return (sum + 1);
}

File: 1446-consecutive-characters/1446-consecutive-characters.py
class Solution(object):
    def maxPower(self, s):
        _max = 0;
        for i, num in enumerate(s):
            savior = i;
            while i < len(s) - 1 and s[i] == s[i + 1]:
                i+=1;
            _max = max(_max, i - savior + 1);
        return (_max);